unicorn_mode/samples/speedtest/rust/src/main.rs

```diff
@@ -87,8 +88,40 @@ fn main() {
     fuzz(input_file).unwrap();
 }
 
+struct FuzzCB;
+
+impl<'a, D> UnicornAflExecutorHook<'a, D> for FuzzCB {
+    fn place_input(
+        &mut self,
+        uc: &mut Unicorn<'a, unicornafl::executor::UnicornFuzzData<D>>,
```

Comment:
If the second parameter is always of type UnicornFuzzData, this should just take D as generic

Reply:
Yes so the type here should be Unicorn<'a, D>

Reply:
Ah wait, you need UnicornFuzzData because it's unicornafl specific, not for unicorn directly? Sad

Reply:
I can’t recall why we need this type in the public interface.

@Evian-Zhang did this refactoring. Could you elaborate a bit more?

Reply:
If I recall it correctly, this is mostly for hook data here: https://github.com/AFLplusplus/unicornafl/blob/6da4c5315333f0d6f87a4a40e5bd3570261dc5ef/src/executor.rs#L107

Without this, we have to do some bad unsafe pointer with D I think because D is shared across all hooks. Generally, it seems a design flaw from Unicorn bindings and could be improved once unicorn-engine-sys is published with stable APIs.